test/Rewards.t.sol

```diff
@@ -0,0 +1,133 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.23;
+
+import "forge-std/Test.sol";
+import "forge-std/console.sol";
+import "../src/NODL.sol";
+import "../src/Rewards.sol";
+import "openzeppelin-contracts/contracts/utils/cryptography/MessageHashUtils.sol";
+
+contract RewardsTest is Test {
+    NODL nodlToken;
+    Rewards rewards;
+    address recipient;
+    uint256 oraclePrivateKey;
+
+    function setUp() public {
+        recipient = address(1);
+        oraclePrivateKey = 0xBEEF;
+        address oracle = vm.addr(oraclePrivateKey);
+
+        nodlToken = new NODL();
+        rewards = new Rewards(address(nodlToken), 1000, 1 days, oracle);
+        // Grant MINTER_ROLE to the Rewards contract
+        nodlToken.grantRole(nodlToken.MINTER_ROLE(), address(rewards));
+    }
+
+    function testSetQuota() public {
+        // Check initial quota
+        assertEq(rewards.rewardQuota(), 1000);
+
+        // Try to set the quota without the role
+        vm.expectRevert();
```

Comment:
We should filter for the precise error here. See my helper here https://github.com/NodleCode/rollup/blob/main/test/__helpers__/AccessControlUtils.sol

Reply:
Done. I had tried: vm.expectRevert(IAccessControl.AccessControlUnauthorizedAccount.selector, address(this), rewards.QUOTA_SETTER_ROLE()) which was not working potentially due to a bug in foundry

Reply:
I think you were missing the`abi.encodeWithSelector` call when you tried :)